app/views/alaveteli_pro/subscriptions/_card.html.erb

```diff
@@ -1,30 +1,35 @@
-<%= card.brand %>
-<%= card.last4 %>
-<%= card.exp_month %>/<%= card.exp_year %>
-<%= card_expiry_message(card.exp_month, card.exp_year) %>
+<div class="settings__item">
+  <div class="settings__item__primary">
+    <span class="card-desc"><strong><%= card.brand %></strong> <%= _('ending in') %> <%= card.last4 %><%= _(', expiry') %> <%= card.exp_month %>/<%= card.exp_year %></span>
+  </div>
+  <div class="settings__item__secondary">
+    <%= form_tag(update_subscription_payment_path) do %>
+
+    <%= hidden_field_tag 'old_card_id', card.id %>
 
-<%= form_tag(update_subscription_payment_path) do %>
+    <script src="https://checkout.stripe.com/checkout.js" class="stripe-button"
+            data-key="<%= AlaveteliConfiguration.stripe_publishable_key %>"
+            data-name="<%= AlaveteliConfiguration.pro_site_name %>"
+            data-panel-label="Update Card Details"
+            data-label="Update Card Details"
+            data-email="<%= current_user.email %>"
+            data-locale="auto"
+            data-allow-remember-me=false
+            data-image="https://s3.amazonaws.com/stripe-uploads/acct_19EbqNIbP0iBLddtmerchant-icon-1479145884111-mysociety-wheel-logo.png"
+            data-zip-code=false>
+    </script>
 
-  <%= hidden_field_tag 'old_card_id', card.id %>
+    <noscript>
+      <div id="error">
+        <p>
+          <%= _('Updating your payment card requires your browser to have ' \
+                'JavaScript enabled.') %>
+        </p>
+      </div>
+    </noscript>
+  </div>
+<%= card_expiry_message(card.exp_month, card.exp_year) %>
```

Comment:
I can't tell if this has been put here intentionally or missed.

I'd say it should probably be bold & red?

![screen shot 2017-10-12 at 15 00 38](https://user-images.githubusercontent.com/282788/31499986-2a213fce-af5e-11e7-9ad6-717c4d3b7ce3.png)

Reply:
It could be better.

Is there a way to check for card expiry in the template without replicating the ```card_expiry_message``` function in the template? Eg can I just say ```if (card_expiry_message(...,...))```?

Reply:
ah, that would be my fault, I should probably have done...

```diff
- def card_expiry_message(month, year)
-  if month == Date.today.month && year == Date.today.year
-    _('Expires soon')
-  end
- end

+ def card_expiry_message(card)
+   if card_expiring?(card)
+     _('Expires soon')
+   end
+ end

+ def card_expiring?(card)
+   if card.exp_month == Date.today.month && card.exp_year == Date.today.year
+     true
+   else
+     false
+   end
+ end
```

...then you could just do `card_expiring?(@card)` in the view (I think), and change the call to `card_expiry_message` to pass in `card` instead of the `card.exp_month, card.exp_year`

Reply:
...or ditch `card_expiry_message` of course, might not be simpler to just check for `card_expiring?` in the view

Reply:
Oops sorry, should have pasted our IRC chat here:

```irc
[16:15:33]  <gareth>	hey what do you mean by https://github.com/mysociety/alaveteli/pull/4282#discussion_r144319914?
[16:15:58]  <gareth>	I guess the question I'm not sure of is why you'd need the conditional?
[16:16:20]  <martin>	oh - so the function for the card expiring is just ``` def card_expiry_message(month, year)
[16:16:20]  <martin>	    if month == Date.today.month && year == Date.today.year
[16:16:20]  <martin>	      _('Expires soon')
[16:16:20]  <martin>	    end
[16:16:20]  <martin>	  end```
[16:16:45]  <martin>	I want to wrap that in something for presention, but I only want the wrapping code to appear when the message does
[16:16:52]  <gareth>	ahh
[16:16:54]  <gareth>	with you
[16:16:58]  <martin>	don't want to put HTML in there^
[16:17:10]  <gareth>	You can put HTML in a helper, thats fine
[16:17:16]  <gareth>	I can help with that
[16:17:25]  <martin>	oh ok, didn't realise
[16:17:25]  <gareth>	what's the HTML you want to end up with?
[16:18:01]  <martin>	```<p class="card__expiring">...</p>``` would do it
[16:18:52]  <gareth>	```content_tag(:p, _('Expires soon'), class: 'card__expiring')```
[16:18:53]  <gareth>	:)
```

The solution in 2457345 looks like its resolved the issue.